src/models/analytics-visualisation.ts

```diff
@@ -45,6 +47,8 @@ interface GraphVisualisation {
   xAxisColumnIndex: number | null;
   forceOrdinal: boolean;
   showAllXAxisLabels: boolean;
+  yMin: number | null;
+  yMax: number | null;
```

Comment:
These types look wrong to me

In the client, we only allow this setting on `line` and `scatter` charts. So this should be reflected here

[play](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgEJwM4QMIAs5RjIDeAUMsggPYA2VUGAXMhmFKAOYDaAugNzlkADwCCQ4Bmy0ArgFsQASRAATCEOYg5AI2jIAPsk00aAijHpIA8lGWg4NZlqq0IcEKZa4qAdxHGAGmISADJwOjRMyE4ubgIAvqSkoJCwiCjBoDj4hMhqkCoYaJhZBERkFGAAngAOEMwA5DSZ9frI9RgIcGAp9R6VALKgGtq6BkYmggNw6oYjUK3j8YnJ0PBIRVB4pblC+cqF6FhbOeXIVbUNrIgA1hDKALRaBC0G9Qg00qzQd4-PS6TnFAAcSgcGquAAahJpPYJF1gFQQMgALzIDIgEo5AzoTbZMB8IA)
```ts
interface BaseChart {
  colors: string[];
  xAxisColumnIndex: number | null;
  forceOrdinal: boolean;
  showAllXAxisLabels: boolean;
}

interface LineChart extends BaseChart {
  type: 'line' | 'scatter';
  yMin: number | null;
  yMax: number | null;
}

interface BarChart extends BaseChart {
  type: 'stacked-bar' | 'clustered-bar';
}

type GraphVisualisation = LineChart | BarChart;
```

in the Joi schema, it'd look like (not validated/tested):
```ts
const baseGraphVisualisationSchema = Joi.object().keys({
  colors: Joi.array().items(Joi.string().example('#202f0c')).required(),
  xAxisColumnIndex: Joi.number().integer().example(null).allow(null)
    .required(),
  forceOrdinal: Joi.boolean().example(false).required(),
  showAllXAxisLabels: Joi.boolean().example(false).default(false),
})

const barChartGraphVisualisationSchema = baseGraphVisualisationSchema.keys({
  type: Joi.string().example('stacked-bar').valid('stacked-bar', 'clustered-bar').required(),
})

const lineChartGraphVisualisationSchema = baseGraphVisualisationSchema.keys({
	type: Joi.string().example('line').valid('line', 'scatter').required(),
  yMin: Joi.number().allow(null).example(10).default(null),
  yMax: Joi.number().allow(null).example(10).default(null),
})

const graphVisualisationSchema = Joi.alternatives().try(
  barChartGraphVisualisationSchema,
  lineChartGraphVisualisationSchema,
)
```